### DISCUSSION TITLE
Net core 2 - trim none not working?

### DISCUSSION THREAD

**Discussion by globtech1** [JoshClose/CsvHelper#1176] [Q&A]
I have a dotnet core 2 code 
![image](https://user-images.githubusercontent.com/39185790/49181629-2c421500-f358-11e8-9970-550ab0ea812d.png)

with the following configuration:
![image](https://user-images.githubusercontent.com/39185790/49181227-1da72e00-f357-11e8-9ec3-d3ea7fb71c1c.png)

and this string gets trimmed:
![image](https://user-images.githubusercontent.com/39185790/49181277-3a436600-f357-11e8-93f2-288b222a071f.png)

See here:
![image](https://user-images.githubusercontent.com/39185790/49181583-0ddc1980-f358-11e8-86d1-89069d89b279.png)

Please am I doing something wrong? I need not to trim the fields. Thanks.

**Comment by JoshClose** [comment]
There must be something else going on. Can you make a failing example? This works:
```cs
void Main()
{
	var s = new StringBuilder();
	s.AppendLine("Id,Name");
	s.AppendLine("1,one ");
	using (var reader = new StringReader(s.ToString()))
	using (var csv = new CsvReader(reader))
	{
		csv.Configuration.TrimOptions = TrimOptions.None;
		var records = csv.GetRecords<Test>().ToList();
		$"'{records[0].Name}'".Dump();
	}
}

public class Test
{
	public int Id { get; set; }
	public string Name { get; set; }
}
```

**Comment by CedricDumont** [comment]
Hi, I have the same issue. it trims end:
- project : .net standard 2.0
- CsvHelper : 9.2.3

```csharp

            var s = new StringBuilder();
            s.AppendLine("Id;Name");
            s.AppendLine("1;\" one \"");
            using (var r = new StringReader(s.ToString()))
            using (var csv = new CsvReader(r))
            {
                csv.Configuration.TrimOptions = TrimOptions.None;
                csv.Configuration.IgnoreQuotes = false;
                csv.Configuration.Delimiter = ";";
                var records = csv.GetRecords<Test>().ToList();
                //$"'{records[0].Name}'".Dump();
                Assert.Equal(" one ", records[0].Name); // fails
            }
```
instead of **" one "** I got **" one"**

I did not have that issue with
- project .net framework 4.7
- csvHelper 7.1.1

remark : downgrading to **9.2.2**, the behaviour is ok

**Comment by JoshClose** [comment]
Can you try the latest? (12.1.1)

**Comment by neilgolatkar** [comment]
Hi Josh, there appears to be a problem with trimming values in quotes with leading spaces.

Such as " one" will trim to " one" instead of "one" like it should.

We updated from 9.1.0 to 12.1.x and noticed this issue.

**Comment by JoshClose** [comment]
Can you please provide a failing unit test? I haven't seen the behavior people are talking about yet. This works fine for me in the version 12.2.1.

```cs
void Main()
{
	var s = new StringBuilder();
	s.AppendLine("Id,Name");
	s.AppendLine("1,\" one \"");
	using (var reader = new StringReader(s.ToString()))
	using (var csv = new CsvReader(reader))
	{
		csv.Configuration.TrimOptions = TrimOptions.InsideQuotes;
		csv.Read();
		csv.ReadHeader();
		csv.Read();
		$"'{csv.GetField(1)}'".Dump();
	}
}
```
